src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -265,27 +280,47 @@ function PostEditor({
           </div>
 
           {!editExisting
-          && (
-            <div className="d-flex flex-row mt-3">
-              <Form.Checkbox
-                name="follow"
-                value={values.follow}
-                onChange={handleChange}
-                onBlur={handleBlur}
-                className="mr-4"
-              >
-                {intl.formatMessage(messages.followPost)}
-              </Form.Checkbox>
-              <Form.Checkbox
-                name="anonymous"
-                value={values.anonymous}
-                onChange={handleChange}
-                onBlur={handleBlur}
-              >
-                {intl.formatMessage(messages.anonymousPost)}
-              </Form.Checkbox>
-            </div>
-          )}
+            && (
+              <div className="d-flex flex-row mt-3">
+                <Form.Group>
+                  <Form.Checkbox
+                    name="follow"
+                    checked={values.follow}
+                    onChange={handleChange}
+                    onBlur={handleBlur}
+                    className="mr-4"
+                  >
+                    {intl.formatMessage(messages.followPost)}
+                  </Form.Checkbox>
+                </Form.Group>
+                {allowAnonymous && (
+                  <Form.Group>
+                    <Form.Checkbox
+                      name="anonymous"
+                      checked={values.anonymous}
+                      onChange={handleChange}
+                      onBlur={handleBlur}
```

Comment:
```suggestion
                      onBlur={handleBlur}
                      aria-describedby="Post anonymously checkbox"
```

Reply:
I don't think this is needed. The ``{intl.formatMessage(messages.anonymousPost)}`` we're passing here is set as the label so it's available for accessibility purposes as a label.